base_layer/key_manager/src/cipher_seed.rs

```diff
@@ -264,6 +281,92 @@ impl CipherSeed {
     }
 }
 
+impl CipherSeed {
+    fn generate_mac(
+        birthday: &[u8],
+        entropy: &[u8],
+        cipher_seed_version: &[u8],
+        salt: &[u8],
+        passphrase: &str,
+    ) -> Result<Vec<u8>, KeyManagerError> {
+        // birthday should be 2 bytes long
+        if birthday.len() != CIPHER_SEED_BIRTHDAY_BYTES {
+            return Err(KeyManagerError::InvalidData);
+        }
+
+        // entropy should be 16 bytes long
+        if entropy.len() != CIPHER_SEED_ENTROPY_BYTES {
+            return Err(KeyManagerError::InvalidData);
+        }
+
+        // cipher_seed_version should be 1 byte long
+        if cipher_seed_version.len() != 1 {
+            return Err(KeyManagerError::InvalidData);
+        }
+
+        // salt should be 5 bytes long
+        if salt.len() != CIPHER_SEED_SALT_BYTES {
+            return Err(KeyManagerError::InvalidData);
+        }
+
+        // we take the first 32 bytes of the generated derived encryption key for MAC generation, see documentation
+        let passphrase_key = Self::generate_domain_separated_passphrase_hash(passphrase, salt)?[..32].to_vec();
+
+        let mac = DomainSeparatedHasher::<Blake256, GenericHashDomain>::new(DOMAIN_SEPARATION_LABEL)
```

Comment:
Using the `MacDomain` from the hasing API:
```suggestion
         // TODO: Replace with generic `HashingDomain::mac_hasher` implementation
        let mac = Self::mac_hasher::<Blake256>(DOMAIN_SEPARATION_LABEL)
```